ocs/agents/aggregator/drivers.py

```diff
@@ -335,9 +336,16 @@ def save_to_block(self, data):
             These must match -- in this instance both the word 'test'.
 
         Args:
-            data (dict): data dictionary from incoming data queue
-
+            data (dict):
+                data dictionary from incoming data queue
+            message_idx (int, optional):
+                message index for the incoming data. If a message index is repeated, the data will be ignored.
         """
+        if message_idx is not None:
+            if message_idx == self.last_message_idx:  # Ignore duplicate messages
+                return
```

Comment:
I think we'd still like to log when this happens.  Maybe in a summary way ("dropped x duplicate frames in the past 1 hour") to not swamp the logs.  But _something_ to help us understand that this happening.